trpcultivate_phenotypes/tests/src/Kernel/Validators/DataFileDelimitedTest.php

```diff
@@ -0,0 +1,343 @@
+   *   
+   *   - A string, human-readable short description of the test scenario.
+   *   - Number of expected colum input (first parameter to the setter method).
+   *   - Strict condition flag input (second parameter to the setter method).
+   *   - Expected values set:
+   *     - number_of_columns: the number of columns to expect.
+   *     - strict: strict comparison flag.
+   *   - Expected exception message for both setter and getter:
+   *    - setter: setter exception message.
+   *    - getter: getter exception message.
+   */
+  public function provideValidatorConfigurationValues() {
+    return [
+      // #0: A zero number of expected column.
+      [
+        'zero columns',
+        0,
+        FALSE,
+        NULL,
+        [
+          'setter' => 'The setter method in ValidDelimitedFile requires an integer value greater than zero.',
+          'getter' => 'Cannot retrieve the values set by the ValidDelimitedFile setter method.'
+        ]
+      ],
+
+      // #1: A valid number.
+      [
+        'valid number',
+        10,
+        TRUE,
+        [
+          'number_of_columns' => 10,
+          'strict' => TRUE
+        ],
+        [
+          'setter' => '',
+          'getter' => ''
+        ]
+      ],
+    ];
+  }
+
+  /**
+   * Test getter method to get expected columns before calling
+   * the setter method first.
+   */
+  public function testGetExpectedColumns() {
+    
+    $exception_caught = FALSE;
+    $exception_message = '';
+    
+    try {
+      $this->validator_instance->getExpectedColumns();
+    } catch (\Exception $e) {
+      $exception_caught = TRUE;
+      $exception_message = $e->getMessage();
+    }
+
+    $this->assertTrue($exception_caught, 'ValidDelimitedFile::getExpectedColumns() method should throw an exception when trying to get expected number of columns before setting them.');
+    $this->assertEquals(
+      $exception_message,
+      'Cannot retrieve the values set by the ValidDelimitedFile setter method.',
+      'Exception message does not match the expected message when trying to get expected number of columns before setting them.'
+    );
+  }
+
+  /**
+   * Test getter method to get expected columns.
+   * 
+   * @dataProvider provideValidatorConfigurationValues
+   */
+  public function testValidatorSetterAndGetter($scenario, $column_numbers_input, $strict_input, $expected, $exception) {
+    
+    // Test the setter method.
+    $exception_caught = FALSE;
+    $exception_message = '';
+
+    try {
+      $this->validator_instance->setExpectedColumns($column_numbers_input, $strict_input);
+    }
+    catch (\Exception $e) {
+      $exception_caught = TRUE;
+      $exception_message = $e->getMessage();
+    }
+
+    $this->assertEquals(
+      $exception_message,
+      $exception['setter'],
+      'Exception message does not match the expected message when trying to set a zero value to number of columns in scenario ' . $scenario
```

Comment:
This is testing more than a zero number of columns, so the message should be more generic. How about:
```suggestion
      'Exception message does not match the expected message when trying to call ValidDelimitedFile::setExpectedColumns() in scenario ' . $scenario
```